spring-cloud-dataflow-platform-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/server/config/cloudfoundry/CloudFoundryTaskPlatformAutoConfiguration.java

```diff
@@ -0,0 +1,132 @@
+	public TaskPlatform cloudFoundryPlatform(
+			CloudFoundryPlatformProperties cloudFoundryPlatformProperties) {
+		List<Launcher> launchers = cloudFoundryPlatformProperties.getAccounts().entrySet().stream().map(
+				e -> createAndSaveCFTaskLauncher(e.getKey(), e.getValue())).collect(Collectors.toList());
+		return new TaskPlatform("Cloud Foundry", launchers);
+	}
+
+	private Launcher createAndSaveCFTaskLauncher(String account,
+			CloudFoundryPlatformProperties.CloudFoundryProperties cloudFoundryProperties) {
+		CloudFoundryDeploymentProperties deploymentProperties = cloudFoundryProperties
+				.getDeployment();
+		if (deploymentProperties == null) {
+			// todo: use server level shared deployment properties
+			deploymentProperties = new CloudFoundryDeploymentProperties();
+		}
+		CloudFoundryConnectionProperties connectionProperties = cloudFoundryProperties
+				.getConnection();
+		try {
+			ConnectionContext connectionContext = DefaultConnectionContext.builder()
+					.apiHost(connectionProperties.getUrl().getHost())
+					.skipSslValidation(connectionProperties.isSkipSslValidation())
+					.build();
+			TokenProvider tokenProvider = PasswordGrantTokenProvider.builder()
+					.username(connectionProperties.getUsername())
+					.password(connectionProperties.getPassword()).build();
+			CloudFoundryClient cloudFoundryClient = ReactorCloudFoundryClient.builder()
+					.connectionContext(connectionContext).tokenProvider(tokenProvider)
+					.build();
+			Version version = cloudFoundryClient.info()
+					.get(GetInfoRequest.builder().build())
+					.map(response -> Version.valueOf(response.getApiVersion()))
+					.doOnNext(versionInfo -> logger.info(
+							"Connecting to Cloud Foundry with API Version {}",
+							versionInfo))
+					.timeout(Duration.ofSeconds(deploymentProperties.getApiTimeout()))
+					.block();
+			RuntimeEnvironmentInfo runtimeEnvironmentInfo = new RuntimeEnvironmentInfo.Builder()
+					.implementationName(CloudFoundryAppDeployer.class.getSimpleName())
+					.spiClass(AppDeployer.class)
+					.implementationVersion(
+							RuntimeVersionUtils.getVersion(CloudFoundryAppDeployer.class))
+					.platformType("Cloud Foundry")
+					.platformClientVersion(
+							RuntimeVersionUtils.getVersion(cloudFoundryClient.getClass()))
+					.platformApiVersion(version.toString()).platformHostVersion("unknown")
+					.addPlatformSpecificInfo("API Endpoint",
+							connectionProperties.getUrl().toString())
+					.build();
+			CloudFoundryOperations cloudFoundryOperations = DefaultCloudFoundryOperations
+					.builder().cloudFoundryClient(cloudFoundryClient)
+					.organization(connectionProperties.getOrg())
+					.space(connectionProperties.getSpace()).build();
+			CloudFoundryAppNameGenerator appNameGenerator = new CloudFoundryAppNameGenerator(
+					cloudFoundryProperties.getDeployment());
+			appNameGenerator.afterPropertiesSet();
+
+			CloudFoundry2630AndLaterTaskLauncher cfTaskLauncher = new CloudFoundry2630AndLaterTaskLauncher(
```

Comment:
I think it's time rename this `CloudFoundry2630AndLaterTaskLauncher` to `CloudFoundryTaskLauncher` in CF deployer 2.x :-)

Reply:
Created https://github.com/spring-cloud/spring-cloud-deployer-cloudfoundry/issues/260